addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -331,6 +331,13 @@ export class SelfOrder extends Reactive {
         if (lineToMerge) {
             lineToMerge.qty += newLine.qty;
             newLine.delete();
+            lineToMerge.set_unit_price(
+                lineToMerge.product_id.get_price(
+                    this.currentOrder.pricelist_id,
+                    lineToMerge.get_quantity(),
+                    lineToMerge.get_price_extra()
+                )
+            );
```

Comment:
```suggestion
            lineToMerge.set_quantity(qty + newLine.qty);
            newLine.delete();
```
I think this would work as the recomputation of the unit price is done in set_quantity